src/commands/domain.rs

```diff
@@ -148,3 +135,196 @@ fn print_existing_domains(domains: &DomainsDomains) -> Result<()> {
+        )
+        .with_message(message.unwrap_or_else(|| "Creating domain...".to_string()));
+    spinner.enable_steady_tick(Duration::from_millis(100));
+
+    Ok(spinner)
+}
+
+async fn create_custom_domain(
+    domain: String,
+    port: Option<u16>,
+    service_name: Option<String>,
+    json: bool,
+) -> Result<()> {
+    let configs = Configs::new()?;
+
+    let client = GQLClient::new_authorized(&configs)?;
+    let linked_project = configs.get_linked_project().await?;
+
+    ensure_project_and_environment_exist(&client, &configs, &linked_project).await?;
+
+    let project = get_project(&client, &configs, linked_project.project.clone()).await?;
+
+    let service = get_service(&linked_project, &project, service_name)?;
+
+    let spinner = (std::io::stdout().is_terminal() && !json)
+        .then(|| {
+            creating_domain_spiner(Some(format!(
+                "Creating custom domain for service {}{}...",
+                service.name,
+                port.map(|p| format!(" on port {}", p)).unwrap_or_default()
+            )))
+        })
+        .and_then(|s| s.ok());
+
+    let is_available = post_graphql::<queries::CustomDomainAvailable, _>(
+        &client,
+        configs.get_backboard(),
+        queries::custom_domain_available::Variables {
+            domain: domain.clone(),
+        },
+    )
+    .await?
+    .custom_domain_available
+    .available;
+
+    if !is_available {
+        bail!("Domain is not available:\n\t{}", domain);
+    }
+
+    let vars = mutations::custom_domain_create::Variables {
+        input: mutations::custom_domain_create::CustomDomainCreateInput {
+            domain: domain.clone(),
+            environment_id: linked_project.environment.clone(),
+            project_id: linked_project.project.clone(),
+            service_id: service.id.clone(),
+            target_port: port.map(|p| p as i64),
+        },
+    };
+
+    let response =
+        post_graphql::<mutations::CustomDomainCreate, _>(&client, configs.get_backboard(), vars)
+            .await?;
+
+    spinner.map(|s| s.finish_and_clear());
+
+    if json {
+        println!("{}", serde_json::to_string_pretty(&response)?);
+        return Ok(());
+    }
+
+    println!("Domain created: {}", response.custom_domain_create.domain);
+
+    if response.custom_domain_create.status.dns_records.is_empty() {
+        // This case should be impossible, but added error handling for safety.
+        //
+        // It can only occur if the backend is not returning the correct data,
+        // and in that case, the post_graphql call should have already errored.
+        bail!("No DNS records found. Please check the Railway dashboard for more information.");
+    }
+
+    println!(
+        "To finish setting up your custom domain, add the following DNS records to {}:\n",
+        &response.custom_domain_create.status.dns_records[0].zone
+    );
+
+    print_dns(response.custom_domain_create.status.dns_records);
```

Comment:
Will there ever be more than 1 DNS record? It might make it clearer to just `print_dns` with a single record at a time in a loop for the very edge case where there are more than 1 record.

Reply:
Yes, if you use a `*` wildcard it will have 2 records.
<img width="779" alt="image" src="https://github.com/user-attachments/assets/9ebbca16-25e1-4948-8f0f-17ceabed6ab8">